commit 139a6b89af2e7728d2c4b1c66af6f03429fa43c4
Author: Kalesh Singh <kaleshsingh@google.com>
Date:   Sun Oct 13 13:45:56 2024 -0700

    ltp: syscalls: fcntl: Clarify default pipe size for previleged user
    
    The test doesn't set the pipe's size so we are actually checking the
    default size. In the case of a privileged user this is determined by
    PIPE_DEF_BUFFERS number of pages in the kernel.
    
    User the kernel_page_size() instead of the userspace page size when
    checking the pipe size on Android's x86_64 16KiB page size emulators.
    
    This is effectively a no-op change on all other targets.
    
    Bug: 366098040
    Test: atest vts_ltp_test_x86_64:syscalls.fcntl05_64bit#syscalls.fcntl05_64bit
    Change-Id: I7eb8839d595fd4ff96e5e19bca91e1228aba9759
    Signed-off-by: Kalesh Singh <kaleshsingh@google.com>

diff --git a/testcases/kernel/syscalls/fcntl/fcntl35.c b/testcases/kernel/syscalls/fcntl/fcntl35.c
index 2d730377b..def5ccef4 100644
--- a/testcases/kernel/syscalls/fcntl/fcntl35.c
+++ b/testcases/kernel/syscalls/fcntl/fcntl35.c
@@ -30,6 +30,9 @@
 
 #include "lapi/fcntl.h"
 #include "tst_test.h"
+#include "pgsize_helpers.h"
+
+#define PIPE_DEF_BUFFERS 16
 
 static int pipe_max_unpriv;
 static int test_max_unpriv;
@@ -47,7 +50,12 @@ static struct tcase {
 static void setup(void)
 {
 	test_max_unpriv = getpagesize();
-	test_max_priv = test_max_unpriv * 16;
+	/*
+	 * The test doesn't set the pipe's size so we are actually checking the default size.
+	 * In the case of a privileged user this is determined by PIPE_DEF_BUFFERS number of
+	 * pages in the kernel.
+	 */
+	test_max_priv = kernel_page_size() * PIPE_DEF_BUFFERS;
 
 	if (!access("/proc/sys/fs/pipe-max-size", F_OK)) {
 		SAFE_FILE_SCANF("/proc/sys/fs/pipe-max-size", "%d",
